lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -1631,6 +1631,13 @@ BlockArgument ClassOp::getArgument(size_t portNumber) {
   return getBodyBlock()->getArgument(portNumber);
 }
 
+bool ClassOp::canDiscardOnUseEmpty() {
+  // ClasseOps are referenced by ClassTypes, and these uses are not
+  // discovereable by the symbol infrastructure. Return false here to prevent
+  // passes like symbolDCE from removing our classes.
+  return false;
```

Comment:
This part makes sense to me.

Reply:
Looking a little, looks like SymbolDCE only walks attributes (`op->getAttrDictionary().walk<WalkOrder::PreOrder>([&](SymbolRefAttr symbolRef)``, is that the reason?

Sounds good to me, we can DCE these ourselves as-needed in a few places.

Reply:
Yep.

So I think eventually we should have a FIRRTL syntax for marking classes as public/private, and then a class can be kept alive by either being public, or being instantiated by an object op. But until then, we might as well be conservative.